x-pack/plugins/observability/public/hooks/use_date_format.ts

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useUiSetting } from '../../../../../src/plugins/kibana_react/public';
```

Comment:
When I tried to mock out useUiSetting I ran into trouble, so I added this extra hook as a seam to allow me to mock it.
The other option I considered was to wrap the test in the Kibana context provider and use mount, but I'd rather rely on shallow if possible. This feels worth it in order to stick with shallow.

Reply:
Do you remember the problems exactly? We can try to find some existing tests that mock `useUiSetting` and see if there's something specific that needs to be done

Reply:
I'll try again and see what the issue was. I do see other tests that mock this but some seem to mock the module and not just that one export so I can try that too. 
Depends on if the alternative test render helper is a better approach since then you don't need to mock the hook (instead the context is mocked).

Reply:
I must have done something wrong the first time around, I tried again now and it works just like mocking out any other hook. Maybe I had the wrong path? Anyway :)